### ISSUE TITLE
blocking when using BlockingWaitStrategy.signalAllWhenBlocking

### ISSUE THREAD

**Issue by xiaoma20082008** [LMAX-Exchange/disruptor#173]
I'm using log4j2 and disruptor-3.3.5 (update to this version to solve this [issues138](https://github.com/LMAX-Exchange/disruptor/issues/138) ).
but found this problem on the picture
![image](https://cloud.githubusercontent.com/assets/10076393/18870265/6d043924-84e2-11e6-9060-a6a840c5fe37.png).

env:openjdk1.7

```
99: "catalina-exec-1" Id=167 WAITING on java.util.concurrent.locks.ReentrantLock$NonfairSync@168be842 
at sun.misc.Unsafe.park(Native Method) 
- waiting on java.util.concurrent.locks.ReentrantLock$NonfairSync@168be842 
at java.util.concurrent.locks.LockSupport.park(LockSupport.java:186) 
at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:834) 
at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:867) 
at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1197) 
at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:214) 
at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:290) 
at com.lmax.disruptor.BlockingWaitStrategy.signalAllWhenBlocking(BlockingWaitStrategy.java:65)

```

**Comment by mikeb01** [comment]
I'm struggling to understand what the actual issue is that you are reporting.  What impact is it having on the system.

**Comment by xiaoma20082008** [comment]
All cpu used nearly 100%,all other request cannot be handled。
 java-version="1.7.0_80"
<os name="Linux" arch="amd64" version="2.6.32-504.30.3.el6.x86_64" available-processors="4" total-physical-memory="8252825600" free-physical-memory="138006528" committed-virtual-memory="5668225024" total-swap-space="0" free-swap-space="0"/>

**Comment by mikeb01** [comment]
I believe that this is an issue with Log4J and not the disruptor.  I would also recommend updating your version of the JDK to the latest release.
